pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -428,6 +432,84 @@ func (c ClusterSpec) IsClusterBackupEnabled() bool {
+	// Contains the settions for the AuthorizationWebhook if EnabledModes contains Webhook
+	AuthorizationWebhookConfiguration *AuthorizationWebhookConfiguration `json:"authorizationWebhookConfiguration,omitempty"`
+
+	AuthorizationConfigurationFile *AuthorizationConfigurationFile `json:"authorizationConfigurationFile,omitempty"`
+}
+type AuthorizationWebhookConfiguration struct {
+	// The secret containing the webhook configuration
+	SecretName string `json:"secretName"`
+	// The secret Key inside the secret
+	SecretKey string `json:"secretKey"`
+	// the Webhook Version, by default "v1"
+	WebhookVersion string `json:"webhookVersion"`
+}
+
+type AuthorizationConfigurationFile struct {
+	// The secret containing the authorizaion configuration
+	SecretName string `json:"secretName"`
+	// The secret Key containing the AuthorizationConfig k8s object
+	SecretKey string `json:"secretKey"`
+	// the path were the secret should be mounted, by default '/etc/kubernetes/authorization-configs'
+	SecretMountPath string `json:"secretMountPath"`
+}
+
+func (c ClusterSpec) GetAuthorizationModesString() string {
+	if c.AuthorizationConfig != nil && c.AuthorizationConfig.EnabledModes != nil {
+		return strings.Join(c.AuthorizationConfig.EnabledModes, ",")
+	}
+
+	return "Node,RBAC"
+}
+
+func (c ClusterSpec) IsWebhookAuthorizationEnabled() bool {
+	if c.AuthorizationConfig == nil || c.AuthorizationConfig.EnabledModes == nil || c.AuthorizationConfig.AuthorizationWebhookConfiguration == nil {
+		return false
+	}
+
+	if !slices.Contains(c.AuthorizationConfig.EnabledModes, "Webhook") {
+		return false
+	}
+
+	if len(c.AuthorizationConfig.AuthorizationWebhookConfiguration.SecretName) == 0 || len(c.AuthorizationConfig.AuthorizationWebhookConfiguration.SecretKey) == 0 {
+		return false
+	}
+	return true
+}
+
+func (c ClusterSpec) GetAuthorizationWebhookVersion() string {
+	if c.AuthorizationConfig != nil && c.AuthorizationConfig.AuthorizationWebhookConfiguration != nil && len(c.AuthorizationConfig.AuthorizationWebhookConfiguration.WebhookVersion) > 0 {
+		return c.AuthorizationConfig.AuthorizationWebhookConfiguration.WebhookVersion
+	}
+
+	return "v1"
+}
+
+func (c ClusterSpec) IsAuthorizationConfigurationFileEnabled() bool {
+	if c.AuthorizationConfig == nil || c.AuthorizationConfig.AuthorizationConfigurationFile == nil {
+		return false
+	}
+
+	if len(c.AuthorizationConfig.AuthorizationConfigurationFile.SecretName) == 0 || len(c.AuthorizationConfig.AuthorizationConfigurationFile.SecretKey) == 0 {
+		return false
+	}
+
+	return true
+}
+
+func (c ClusterSpec) GetAuthorizationConfigurationMountPath() string {
+	if c.AuthorizationConfig != nil && c.AuthorizationConfig.AuthorizationConfigurationFile != nil && len(c.AuthorizationConfig.AuthorizationConfigurationFile.SecretMountPath) > 0 {
+		return c.AuthorizationConfig.AuthorizationConfigurationFile.SecretMountPath
+	}
+
+	return "/etc/kubernetes/authorization-configs"
+}
```

Comment:
The mount path should not be determined in pkg/apis, it's an implementation detail of KKP that has no place being in our stable public API definitions. Or to put it differently, there is no reason for anyone outside of KKP to know where we mount something in the apiserver Deployment.